src/components/manager/ManagerIngestionContent.js

```diff
@@ -13,233 +13,156 @@ import {
-            // TODO: GUI error message
-            return;
-        }
-
-        const serviceInfo = this.props.servicesByID[this.state.selectedWorkflow.serviceID];
-        const tableID = this.state.selectedTable.split(":")[2];
-
-        this.props.submitIngestionWorkflowRun(serviceInfo, tableID, this.state.selectedWorkflow,
-            this.state.inputs, withBasePath("admin/data/manager/runs"), history);
-    }
-
-    getStepContents() {
-        const getTableName = (serviceID, tableID) =>
-            (((this.props.tablesByServiceID[serviceID] || {}).tablesByID || {})[tableID] || {}).name;
-
-        const formatWithNameIfPossible = (name, id) => name ? `${name} (${id})` : id;
-
-        switch (this.state.step) {
-            case STEP_WORKFLOW_SELECTION: {
-                const workflows = this.props.workflows
-                    .filter(w => w.data_type === (this.state.selectedTable
-                        ? this.state.selectedTable.split(":")[1] : null))
-                    .map(w => <WorkflowListItem key={w.id}
-                                                workflow={w}
-                                                selectable={true}
-                                                onClick={() => this.handleWorkflowClick(w)} />);
-
-                return <Form labelCol={FORM_LABEL_COL} wrapperCol={FORM_WRAPPER_COL}>
-                    <Form.Item label="Table">
-                        <TableTreeSelect onChange={table => this.setState({selectedTable: table})}
-                                         value={this.state.selectedTable}/>
-                    </Form.Item>
-                    <Form.Item label="Workflows">
-                        {this.state.selectedTable
-                            ? <Spin spinning={this.props.workflowsLoading}>
-                                {this.props.workflowsLoading
-                                    ? <Skeleton/>
-                                    : <List itemLayout="vertical">{workflows}</List>}
-                            </Spin>
-                            : <Empty image={Empty.PRESENTED_IMAGE_SIMPLE}
-                                     description="Select a table to see available workflows"/>
-                        }
-                    </Form.Item>
-                </Form>;
+import {workflowsStateToPropsMixin} from "../../propTypes";
+import RunSetupWizard from "./RunSetupWizard";
+import RunSetupInputsTable from "./RunSetupInputsTable";
+
+
+const IngestWorkflowSelection = ({values, setValues, handleWorkflowClick}) => {
+    const {workflows, workflowsLoading} = useSelector(workflowsStateToPropsMixin);
+    const {selectedTable} = values;
+
+    const workflowItems = workflows.ingestion
+        .filter(w => w.data_type === (selectedTable ? selectedTable.split(":")[1] : null))
+        .map(w =>
+            <WorkflowListItem
+                key={w.id}
+                workflow={w}
+                selectable={true}
```

Comment:
Just for my understanding, what makes some WorkflowListItems selectable or not?

Reply:
in the workflow tab, there are lists of workflows which cannot be clicked, whereas here clicking on one selects it as the ingestion workflow. probably, this should just be done based on the presence of a handler function rather than a separate attr, but this is just ported from old code right now.

Reply:
removed it in favour of just detecting the presence of a click handler